sofar/sofar.py

```diff
@@ -0,0 +1,2125 @@
+    the json files. To get a list of all currently available SOFA conventions
+    and their paths see :py:func:`~sofar.list_conventions`.
+
+    .. note::
+        If the official convention contain errors, calling this function might
+        break sofar. Be sure that you want to do this.
+
+    Parameters
+    ----------
+    conventions_path : str, optional
+        Path to the folder where the conventions are saved. The default is
+        ``None``, which saves the conventions inside the sofar package.
+        Conventions saved under a different path can not be used by sofar.
+    """
+
+    # url for parsing and downloading the convention files
+    url = ("https://github.com/sofacoustics/API_MO/tree/"
+           "master/API_MO/conventions")
+    url_raw = ("https://raw.githubusercontent.com/sofacoustics/API_MO/"
+               "master/API_MO/conventions")
+    ext = 'csv'
+
+    print(f"Reading SOFA conventions from {url} ...")
+
+    # get file names of conventions from the SOFA Matlab/Octave API
+    page = requests.get(url).text
+    soup = BeautifulSoup(page, 'html.parser')
+    conventions = [os.path.split(node.get('href'))[1]
+                   for node in soup.find_all('a')
+                   if node.get('href').endswith(ext)]
+
+    # Loop conventions
+    updated = False
+    for convention in conventions:
+
+        # exclude these conventions
+        if convention.startswith(("General_", "GeneralString_")):
+            continue
+
+        if conventions_path is None:
+            filename_csv = os.path.join(
+                os.path.dirname(__file__), "conventions", convention)
+        else:
+            filename_csv = os.path.join(conventions_path, convention)
+
+        filename_json = filename_csv[:-3] + "json"
+
+        # download SOFA convention definitions to package diretory
+        data = requests.get(url_raw + "/" + convention)
+
+        # check if convention needs to be added or updated
+        update = False
+        if not os.path.isfile(filename_csv):
+            update = True
+            updated = f"- added new convention: {convention}"
+        else:
+            with open(filename_csv, "rb") as file:
+                data_current = file.readlines()
+            if b"".join(data_current) != data.content:
+                update = True
+                updated = f"- updated existing convention: {convention}"
+
+        # update convention
+        if update:
+            with open(filename_csv, "wb") as file:
+                file.write(data.content)
+            print(updated)
+
+        # convert SOFA conventions from csv to json
+        convention_dict = _convention_csv2dict(filename_csv)
+
+        with open(filename_json, 'w') as file:
+            json.dump(convention_dict, file)
+
+    if updated:
+        print("... done.")
+    else:
+        print("... conventions already up to date.")
+
+
+def list_conventions(verbose=True, return_type=None):
```

Comment:
Good point. I splitted this in a private and a public function. Otherwise we would have ended up with a get_conventions functions with a switch to disable getting the conventions - rofl